test/unit/specs/Datepicker/Datepicker.spec.js

```diff
@@ -339,9 +340,9 @@ describe('Datepicker mounted', () => {
   })
 
   it('emits focus', async () => {
-    const input = wrapper.find('input')
-    await input.trigger('focusin')
-
+    // See https://github.com/vuejs/vue-test-utils/issues/1932
+    // await input.trigger('focus')
+    await wrapper.element.dispatchEvent(new Event('focusin'))
     expect(wrapper.emitted('focus')).toBeTruthy()
```

Comment:
@mst101 hey i got a little confused about this test here. Before the update this test worked but now it will break if it was like before with 
```
 const input = wrapper.find('input')

    // See https://github.com/vuejs/vue-test-utils/issues/1932
    // await input.trigger('focus')
    await input.element.dispatchEvent(new Event('focus'))

    expect(wrapper.emitted('focus')).toBeTruthy()
```
As i understand the code the focus event on the input field will never trigger the focus event on the datepicker so it could never succeed. 🤔 
So i changed it to trigger the focusin event on the datepicker to emit a focus event.

Is this correct?

Reply:
Hmm, I think that test would be better written as:

```
  it('emits focus', async () => {
    const anyElement = wrapper.find('input')

    await anyElement.trigger('focusin')
    expect(wrapper.emitted('focus')).toBeTruthy()
  })

```

Likewise, the next test should be:

```
  it('emits blur', async () => {
    const anyElement = wrapper.find('input')
    await anyElement.trigger('focusin')
    await anyElement.trigger('focusout')

    expect(wrapper.emitted('blur')).toBeTruthy()
  })

```

Reply:
The dateinput doesn't have a `focusin` event. Thats why i changed it to the wrapper element (Datepicker.vue)

Reply:
Yes, we decoupled the 'focus' event from the input field. So, the datepicker will emit a 'focus' event when ***any element*** is focused. e.g.  ` const anyElement = wrapper.find('button')` works too.

See `handleFocusIn()` > `isActive watcher` > `datepickerIsActive()`